@pauliescanlon/gatsby-mdx-embed/src/components/CodePen/CodePen.tsx

```diff
@@ -1,6 +1,6 @@
 import React, { FunctionComponent } from 'react'
 
-export interface ICodePenProps {
```

Comment:
I'd prefer to keep this an interface as some of the comments such as YouTube, Vimeo, Twitch will share some common props and eventually i'd like to create an interface for these component to extend. I don't think that's possible when using type is it?

Also i prefer to name the props interface to it's named after the component its used in rather than the generic name Props

Reply:
You can also extend types by doing

```ts
type A = {
  a1: string;
  a2: number;
};

type B = {
  b1: boolean;
  b2: string;
};

Type C = A & B &  {
  c: () => void;
};
```

This way `C` will have everything from `A` & `B` _and_ will have  an extra `c`.

---

Since there's only one `Props`, I think it's more verbose to always add the name of the component.
That's why I removed it.

Reply:
Was just having a read up on type vs interface. Seems like TypeScript has changed these rules since i first started using it. Even so i think i'd prefer to keep them as interfaces. It's just my preference really.

Reply:
I've always read that it's better to use `type` instead of `interface` for component props.
Can't find the explanation anymore tho.

If you look at [all my PRs I did for `codesandbox-client`](https://github.com/codesandbox/codesandbox-client/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3AMichaelDeBoey), you'll see I've been switching to `type` there too.

Reply:
I might loop in my mate Ben, he's a bit more of an expert with TypeScript... 

@bnwan can i pick your brains about `interface` vs `types`

Reply:
They are quite similar, even though there are some differences. I mostly prefer interfaces as I think they are much more readable. The important thing is to stay consistent

Reply:
@bnwan Thanks mate! that's good enough for me. 

There seems to be little difference technically between types and interfaces but my preference is interfaces too so unless there's a full on reason to change them i'd like to keep them as they are.

Reply:
I'll make them `interfaces` again 🙂

Reply:
i think i'd prefer them to keep their name too. Just calling them Props might lead to issues further down the road.

Reply:
I do it in all my projects like this.
We don't export them anymore (and don't need to, cause we can derive them by using `ComponentProps`), so there's only 1 `Props` in each file.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3e1e03d0d43d9cf569c93e27c8763e2e94a6674c/types/react/index.d.ts#L779-L784

Reply:
I understand the approach it's a preference thing, and my preference is to leave them as named.

Reply:
OK I'll change them back then 🙂